### ISSUE TITLE
ValueError

### ISSUE THREAD

**Issue by staceysara** [HanGuo97/lq-lora#6]
https://github.com/HanGuo97/lq-lora/blob/main/models/allocation_utils.py#L277 I got this error while using backend: scipy. 
i0 is 20 and i1 is 19. In this case, how can I solve this issue?

**Comment by HanGuo97** [comment]
Except for some early experiments, we used `gurobi` for our experiments. It's a commercial solver, but provides free license for academic users. Would this address the problem?

**Comment by staceysara** [comment]
Okay! I'll try to use gurobi then, thanks!

**Comment by rG223** [comment]
Hi, i encounter a similar error at https://github.com/HanGuo97/lq-lora/blob/c2424b3adc27197815da1ac9e1304565168d824d/models/lora_utils.py#L527. 

Here is my script, and I already used gurobi. I was wondering how to fix this error? Thank you for checking out my comment.

```
task="qnli"
index=0

for data in "None"
do

for budget in "3"
do

echo "GPU ${index}: data=${data}, budget=${budget}"

WANDB_PROJECT="20221007" CUDA_VISIBLE_DEVICES="${index}" python run_glue.py \
    --model_name_or_path roberta-large \
    --output_dir output_glue_lora_20230924_ranks64_${task}_${data}_${budget} \
    --task_name ${task} \
    \
    --bf16 True \
    --tf32 True \
    --do_train \
    --do_eval \
    --max_seq_length 128 \
    --per_device_train_batch_size 32 \
    --learning_rate 2e-5 \
    --num_train_epochs 10 \
    --save_strategy "steps" \
    --save_steps 0.5 \
    --logging_steps 1 \
    \
    --lora_num_ranks 64 \
    --lora_model_name "roberta-large/lora/${data},budget=${budget}" \
    --lora_dropout 0.0 \
    --lora_config "lora" &

index=$(($index+1))

done

wait

done

```

**Comment by rG223** [comment]
Another issue is the use of `--evaluation_strategy "epoch"` in the script `glue_lora.sh`. The file run_glue.py does not accept the `evaluation_strategy` argument. Is this the expected behavior for the current version of `run_glue.py`? When I removed the line containing `--evaluation_strategy "epoch"`, the error disappeared.

**Comment by HanGuo97** [comment]
Hi, thanks for giving it a try!

- I think the line you pointed to is self-explanatory?

- Regarding the second question, the `run_glue` file is somewhat outdated, which could be why you’re experiencing those issues. It might also explain the first problem you encountered.

**Comment by rG223** [comment]
> Hi, thanks for giving it a try!
> 
> * I think the line you pointed to is self-explanatory?
> * Regarding the second question, the  file is somewhat outdated, which could be why you’re experiencing those issues. It might also explain the first problem you encountered.`run_glue`

So should I add the following code to avoid the issue?
```
    for name, module in model.named_modules():
        if isinstance(module, torch.nn.Linear):
            module.bias = None
```
The loaded model contains bias, but it is conflicting with assertion.

**Comment by HanGuo97** [comment]
Hmm, I think when we first developed this, we didn’t have bias in mind (either intentionally or not). My coding style is such that I tend to be more conservative in terms of assumption checking. As such, it might be fine to remove the assertions—at your own risk. I briefly thought about this, and that some of the math _might_ be different once bias is introduced.

**Comment by rG223** [comment]
I got it. Thanks for your suggestions.

**Comment by HanGuo97** [comment]
Hope that helps!

**Comment by rG223** [comment]
Unfortunately, when I don’t consider bias of the model, I get another error during fine-tuning phase:` torch._dynamo.exc.Unsupported: cache_size_limit reached.` This doesn’t seem to be a cache_size limit issue because exporting `TORCHDYNAMO_CACHE_SIZE_LIMIT=10000` still doesn’t work. How can I fix this error if I want to run the provided script?

**Comment by HanGuo97** [comment]
Hmm, what if you keep increasing the cache size? This might be because you use variable batch sizes, since the compilation is shape-dependent, so changing the batch size results in different shapes.

Alternatively, you can use dynamic compilation. Not sure if we support this, but it's worth a try.

**Comment by rG223** [comment]
The use of batch sizes is same, `--per_device_train_batch_size 32`.
By the way, I also set `padding = "max_length"`.
Yes I currently use dynamic compilation, while the GPU memory is increased with the fine-tuning phase, and then raise `torch._dynamo.exc.Unsupported: cache_size_limit reached`.

**Comment by HanGuo97** [comment]
If you keep running into challenges with compile, you can also turn that off. It'll be slower though.
